lib/puppet/type/cron.rb

```diff
@@ -109,28 +109,23 @@ def alphacheck(value, ary)
       false
     end
 
-    def should_to_s(newvalue = @should)
-      if newvalue
-        newvalue = [newvalue] unless newvalue.is_a?(Array)
-        if self.name == :command or newvalue[0].is_a? Symbol
-          newvalue[0]
-        else
-          newvalue.join(",")
+    def should_to_s(value = @should)
+      if value
+        if name == :command || value.is_a?(Array) && value[0].is_a?(Symbol)
+          value = value[0]
         end
+        super(value)
       else
         nil
       end
     end
 
-    def is_to_s(currentvalue = @is)
-      if currentvalue
-        return currentvalue unless currentvalue.is_a?(Array)
-
-        if self.name == :command or currentvalue[0].is_a? Symbol
-          currentvalue[0]
-        else
-          currentvalue.join(",")
+    def is_to_s(value = @is)
+      if value
+        if name == :command || value.is_a?(Array) && value[0].is_a?(Symbol)
```

Comment:
I prefer to use parenthesis rather than relying on everyone to remember that `&&` has higher precedence than `||`.